packages/keyring-controller/src/KeyringController.ts

```diff
@@ -895,7 +899,7 @@ export class KeyringController extends BaseController<
    * operation completes.
    */
   async persistAllKeyrings(): Promise<boolean> {
-    return this.#withControllerLock(async () => this.#updateVault());
+    return this.#persistOrRollback(async () => true);
   }
```

Comment:
This ensures that all vault update operations triggered by clients can also fail safely